fabric-item-api-v1/src/main/java/net/fabricmc/fabric/mixin/item/AbstractFurnaceBlockEntityMixin.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.mixin.item;
+
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.Shadow;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.Redirect;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfo;
+
+import net.minecraft.block.BlockState;
+import net.minecraft.block.entity.AbstractFurnaceBlockEntity;
+import net.minecraft.item.ItemStack;
+import net.minecraft.util.collection.DefaultedList;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.world.World;
+
+@Mixin(AbstractFurnaceBlockEntity.class)
+public abstract class AbstractFurnaceBlockEntityMixin {
+	@Shadow
+	protected DefaultedList<ItemStack> inventory;
+	private static DefaultedList<ItemStack> capturedInventory;
+
+	@Inject(method = "tick", at = @At("HEAD"))
+	private static void getStackCraftingRemainder(World world, BlockPos pos, BlockState state, AbstractFurnaceBlockEntity blockEntity, CallbackInfo ci) {
+		capturedInventory = ((AbstractFurnaceBlockEntityMixin) (Object) blockEntity).inventory;
+	}
+
+	@Redirect(method = "tick", at = @At(value = "INVOKE", target = "Lnet/minecraft/item/ItemStack;isEmpty()Z", ordinal = 2))
+	private static boolean setStackCraftingRemainder(ItemStack itemStack) {
+		capturedInventory.set(1, itemStack.getRecipeRemainder().copy());
+		return true;
+	}
```

Comment:
This fixes the bug that overwrites entire stacks, and uses the vanilla way to put the remainder.

PS. Maybe if there is no space in the fuel slot the remainder should be dropped? 🤔

```suggestion
	private static boolean setStackCraftingRemainder(ItemStack stack) {
		//Gets the remainder.
		ItemStack remainder = stack.getRecipeRemainder().copy();

		//Decrements the stack like vanilla do.
		stack.decrement(1);

		//Insert the remainder if the stack is empty.
		if (stack.isEmpty()) {
			capturedInventory.set(1, remainder);
		}

		return true;
	}
```